scripts/convert_lit_checkpoint.py

```diff
@@ -0,0 +1,238 @@
+        "model.layers.{}.mlp.up_proj.weight": "transformer.h.{}.mlp.fc_2.weight",
+        "model.layers.{}.mlp.down_proj.weight": "transformer.h.{}.mlp.proj.weight",
+        "model.norm.weight": "transformer.ln_f.weight",
+        "lm_head.weight": "lm_head.weight",
+    }
+
+    for name, param in lit_weights.items():
+        # handle name
+        if "transformer.h" in name and not name.endswith(".attn.attn.weight"):
+            from_name, number = layer_template(name, 2)
+            to_name = get_to_name(from_name, weight_map)
+            if to_name is None:
+                continue
+            to_name = to_name.format(number)
+        elif name.endswith(".attn.attn.weight"):
+            from_name, number = layer_template(name, 2)
+            q = "model.layers.{}.self_attn.q_proj.weight".format(number)
+            k = "model.layers.{}.self_attn.k_proj.weight".format(number)
+            v = "model.layers.{}.self_attn.v_proj.weight".format(number)
+        else:
+            to_name = get_to_name(name, weight_map)
+
+        # handle param
+        if name.endswith(".attn.attn.weight"):
+            qkv = load_param(param)
+            qp, kp, vp = tensor_split(qkv, config, "llama")
+            for to_name, _param in zip((q, k, v), (qp, kp, vp)):
+                if saver is not None:
+                    param = saver.store_early(_param)
+                state_dict[to_name] = param
+        else:
+            param = load_param(param)
+            if saver is not None:
+                param = saver.store_early(param)
+            state_dict[to_name] = param
+
+
+def tensor_split(param: Union[torch.Tensor, NotYetLoadedTensor], config: Config, model_name: str) -> torch.Tensor:
+    if model_name != "llama":
+        raise NotImplementedError(f"{model_name}")
+    else:
+
+        def middle(start, head_size) -> int:
+            return start + head_size
+
+        def end(start, head_size) -> int:
+            return start + int(head_size * 2)
+
+        head_size = config.head_size
+        nobs = param.shape[0] + 1
+        nsplits = len(("q", "k", "v"))
+        splice_starts = range(head_size, nobs, head_size * nsplits)
+
+        splices = [(start, middle(start), end(start)) for start in splice_starts]
+
+    qc = ()
+    kc = ()
+    vc = ()
+
+    for splice in splices:
+        qs, ks, vs = splice
+        qc += (param[qs - head_size : qs, :],)
+        kc += (param[qs:ks, :],)
+        vc += (param[ks:vs, :],)
+
+    q = torch.cat(qc)
+    k = torch.cat(kc)
+    v = torch.cat(vc)
+
+    return q, k, v
+
+
+def get_to_name(lit_key_name: str, weight_map: Dict[str, str]) -> str:
+    for k, v in weight_map.items():
+        if lit_key_name == v:
+            return k
+
+
+@torch.inference_mode()
+def convert_lit_checkpoint(
+    *,
+    checkpoint_dir: Path = Path("checkpoints/tiiuae/falcon-7b"),
+    checkpoint_name: str = "lit_model_finetuned.pth",
```

Comment:
Since the checkpoint name is different depending on the script and if it's not an end-of-training checkpoint this should be made a required argument (not optional) to avoid silently converting the wrong checkpoint

Reply:
implemented in [c874fa2](https://github.com/Lightning-AI/lit-gpt/pull/273/commits/c874fa23187a324f6161283de2020786c4fd0d11)